sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

```diff
@@ -43,18 +53,24 @@
      *
      * @param key The key with which the specified value should be associated.
      * @param value The value to be associated with the specified key.
-     * @throws IllegalArgumentException If {@code key} is {@code null}.
+     * @throws NullPointerException If {@code key} or {@code value} is {@code null}.
      */
     public Context(Object key, Object value) {
         this.parent = null;
         this.key = Objects.requireNonNull(key, "'key' cannot be null.");
-        this.value = value;
+        this.value = Objects.requireNonNull(value, "'value' cannot be null.");
```

Comment:
`null` should be allowed as a valid value

Reply:
will undo it